org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/BaseDiagnosticsHandler.java

```diff
@@ -170,6 +170,13 @@ private void collectNonJavaProblems(List<Diagnostic> diagnostics, boolean isDiag
 				}
 				list.add(marker);
 			}
+			list.removeIf(marker -> JavaLanguageServerPlugin.getPreferencesManager().getClientPreferences().excludedMarkerTypes().stream().anyMatch(t -> {

```

Comment:
Why is this needed ? At the very bottom of `collectNonJavaProblems(..)` there is a call to : https://github.com/eclipse/eclipse.jdt.ls/blob/c254d4226145152734100fe8120f71fd58876890/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/BaseDiagnosticsHandler.java#L181 . The method handles `list` filtering via. `excludedMarkerTypes`, so why do it here when it's going to happen anyways ?